### ISSUE TITLE
Missing data in Query result when selecting single boolean field.

### ISSUE THREAD

**Issue by jnehlmeier** [eclipse-ee4j/eclipselink#716]
Consider Entity 
```java
class Account {
  boolean isActive;
}
```
and corresponding table containing multiple rows with `true` or `false` in the `isActive` column.

The query
```sql
SELECT a.isActive FROM Account a
```
now only returns rows whose `isActive` field contains `false`. **All `true` rows are missing**. The SQL Query is generated correctly, so some EclipseLink code removes these rows from the final result. You can also use a query with a complex `WHERE` and/or `GROUP BY` clause. Same issue as long as you only select a single boolean field.

As a workaround you could do
```sql
SELECT a.isActive, a.isActive FROM Account a
```
and force an `Object[]` to also get the rows containing `true` as well.

**Comment by rfelcman** [comment]
Sorry but how `@Id` is implemented in provided Entity `Account`. Could You please provide complete code there?

**Comment by jnehlmeier** [comment]
@rfelcman All our entities extend a base class (`@MappedSuperclass`) which defines the id as

```java
  @Id
  @Column(name = "id")
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  @Nullable
  private Long id;
```

Don't know if it matters but this base class also has a `@Version Integer version` field for optimistic locking.

**Comment by rfelcman** [comment]
Fix for this bug was merged to master branch (3.0) and 2.7.7 branch/version. I am closing this bug now. If You will discover another trouble related with this bug, feel free to reopen it.
Nightly build with this fix is available at https://www.eclipse.org/eclipselink/downloads/nightly.php .
